widget/widgetAssets/js/widgetUtility.service.js

```diff
@@ -0,0 +1,96 @@
+/* Copyright start
+    MIT License
+    Copyright (c) 2024 Fortinet Inc
+Copyright end */
+'use strict';
+
+(function () {
+  angular
+    .module('cybersponse')
+    .factory('widgetUtilityService', widgetUtilityService);
+
+  widgetUtilityService.$inject = ['$q', '$http', '$injector', '$interpolate', 'toaster'];
```

Comment:
suggestion: this is a common code and can be a part of widgetUtilities widget and all widget can refer it